canvas/views/views.py

```diff
@@ -78,3 +78,26 @@ def events_options_view(request):
     return render(request, 'canvas/course_event_options.html', {
         'events': course.events.all(),
     })
+
+
+def create_event_view(request, pk):
+    course = get_object_or_404(CanvasCourse, pk=pk)
+    course.is_instructor(request.user)
+    if request.method == 'POST':
+        if course.is_instructor(request.user):
+            form = CreateEventForm(request.POST)
+            if form.is_valid():
+                new_event = form.save(commit=False)
+                new_event.course = course
+                new_event.save()
+                messages.add_message(request, messages.SUCCESS, 'Event created successfully.')
+        else:
+            messages.add_message(request, messages.ERROR,
```

Comment:
permissions should be handled with permission classes or functions. You should not manually do this.